### PR TITLE
Add ability to insert bitmap images from in-memory copies.

### PR DESCRIPTION
**Pull request by rossigee**
This patch adds a new 'insert_bitmap_data' function on the Worksheet to complement the 'insert_bitmap' function. As the existing 'insert_bitmap' function depends on a filename as one if it's arguments, the new function allows programs that have a copy of the bitmap in memory to add it to the worksheet without first having to write it to file.

### CODE REVIEW

File: tests/test_bitmaps.py [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+#coding:utf-8
+# Author:  Ross Golder --<ross@golder.org>
+# Purpose: test_bitmaps
+# Created: 23/04/2014
+# Copyright (C) 2014, Ross Golder
+# License: BSD licence
```
Comment (cjw296) [inline tests/test_bitmaps.py]: can't land it with this stuff I'm afraid, please don't put licensing stuff in your commits, if you're not happy to give over your code, you're not allowed to play, I'm afraid...

File: tests/test_bitmaps.py [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+class TestBitmaps(unittest.TestCase):
+    def test_create_bitmap_from_memory(self):
+        book = xlwt.Workbook()
+        sheet = book.add_sheet('bitmap from memory')
+        
+        with open ("tests/boats.bmp", "r") as bmpfile:
+            bmpdata = bmpfile.read()
+            sheet.insert_bitmap_data(bmpdata, 1, 1)
+        
+        self.assertTrue(True)
```
Comment (cjw296) [inline tests/test_bitmaps.py]: This isn't a useful assertion. Please write tests that demonstrate things work.

File: tests/test_bitmaps.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+    def test_create_bitmap_from_file(self):
+        book = xlwt.Workbook()
+        sheet = book.add_sheet('bitmap from file')
+        
+        sheet.insert_bitmap("tests/boats.bmp", 1, 1)
+        
+        self.assertTrue(True)
```
Comment (cjw296) [inline tests/test_bitmaps.py]: ditto

File: tests/test_bitmaps.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+if __name__=='__main__':
+    unittest.main()
```
Comment (cjw296) [inline tests/test_bitmaps.py]: no need for this thanks.

File: xlwt/Bitmap.py [hunk 5]
```diff
@@ -260,3 +267,14 @@ def __init__(self, filename):
...
+    def __init__(self, filename):
+        """Insert a 24bit bitmap image in a worksheet. The main record required is
+        IMDATA but it must be proceeded by a OBJ record to define its position.
+
+        """
+        BiffRecord.__init__(self)
+
+        self.width, self.height, self.size, data = _process_bitmap(filename)
+        self._write_imdata(data)
+
```
Comment (cjw296) [inline xlwt/Bitmap.py]: This doesn't feel right. Not sure we need another whole new class. 
Could we just have one class that takes data and change the calling code of the one that uses the filename?

**Comment by cjw296** [comment]
If you're still interested, please can you look at the comments in this pull request and address them before rebasing (we've landed python 3 support) and trying again?

File: tests/test_bitmaps.py (hunk 1 above)
Comment (rossigee) [inline tests/test_bitmaps.py]: I just copied this over from another test and modified it. I didn't add this. I really don't care if my name's not on the patch.

File: tests/test_bitmaps.py (hunk 2 above)
Comment (rossigee) [inline tests/test_bitmaps.py]: Suggestions as to how I would do that welcome.

**Comment by rossigee** [comment]
I'll try to have a look some time.

**Pull request closed without merging**